python/lance/tests/test_api.py

```diff
@@ -50,7 +52,7 @@ def test_write_categorical_values(tmp_path: Path):
     df = pd.DataFrame({"label": ["cat", "cat", "dog", "person"]})
     df["label"] = df["label"].astype("category")
     table = pa.Table.from_pandas(df)
-    write_table(table, tmp_path / "test.lance")
+    ds.write_dataset(table, tmp_path / "test.lance", format=LanceFileFormat())
```

Comment:
what's the difference between this API and lance.write_dataset API?

Reply:
`ds.write_dataset()` does not write the "versioned" layout.